ports/esp8266/esp_espnow.c

```diff
@@ -0,0 +1,255 @@
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(espnow_deinit_obj, espnow_deinit);
+
+STATIC mp_obj_t espnow_on_send(size_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        return send_cb_obj;
+    }
+
+    send_cb_obj = args[0];
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(espnow_on_send_obj, 0, 1, espnow_on_send);
+
+STATIC mp_obj_t espnow_on_recv(size_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        return recv_cb_obj;
+    }
+
+    recv_cb_obj = args[0];
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(espnow_on_recv_obj, 0, 1, espnow_on_recv);
+
+// pmk(primary_key)
+STATIC mp_obj_t espnow_pmk(mp_obj_t key) {
+    uint8_t key_buf[ESP_NOW_KEY_LEN];
+    _get_bytes(key, ESP_NOW_KEY_LEN, key_buf);
+    ESPNOW_EXCEPTIONS(esp_now_set_kok(key_buf, ESP_NOW_KEY_LEN));
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(espnow_pmk_obj, espnow_pmk);
+
+// lmk(peer_mac, local_key)
+STATIC mp_obj_t espnow_lmk(mp_obj_t addr, mp_obj_t key) {
+    mp_uint_t addr_len;
+    u8 *addr_buf = (u8 *)mp_obj_str_get_data(addr, &addr_len);
+    if (addr_len != ESP_NOW_ETH_ALEN) mp_raise_ValueError("addr invalid");
+    
+    if (key == mp_const_none) {
+        ESPNOW_EXCEPTIONS(esp_now_set_peer_key(addr_buf, NULL, 0));
+    } else {
+        u8 key_buf[ESP_NOW_KEY_LEN];
+        _get_bytes(key, ESP_NOW_KEY_LEN, key_buf);
+        ESPNOW_EXCEPTIONS(esp_now_set_peer_key(addr_buf, key_buf, ESP_NOW_KEY_LEN));
+    }
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(espnow_lmk_obj, espnow_lmk);
+
+// add_peer(peer_mac, [local_key])
+STATIC mp_obj_t espnow_add_peer(size_t n_args, const mp_obj_t *args) {
+    u8 peer_addr[ESP_NOW_ETH_ALEN];
+    _get_bytes(args[0], ESP_NOW_ETH_ALEN, peer_addr);
+    u8 *lmk_buf = NULL;
+
+    if (n_args > 1) {
+        mp_uint_t lmk_len;
+        lmk_buf = (u8 *)mp_obj_str_get_data(args[1], &lmk_len);;
+        if (lmk_len != ESP_NOW_KEY_LEN) mp_raise_ValueError("key invalid");
+    }
+
+    uint8_t channel = 0;
+
+    ESPNOW_EXCEPTIONS(esp_now_add_peer(peer_addr, ESP_NOW_ROLE_COMBO, channel, lmk_buf, ESP_NOW_KEY_LEN));
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(espnow_add_peer_obj, 1, 2, espnow_add_peer);
+
+// del_peer(peer_mac)
+STATIC mp_obj_t espnow_del_peer(mp_obj_t addr) {
+    mp_uint_t addr_len;
+    u8 *addr_buf = (u8 *)mp_obj_str_get_data(addr, &addr_len);
+    if (addr_len != ESP_NOW_ETH_ALEN) mp_raise_ValueError("addr invalid");
+    ESP_NOW_PEER_EXIST(addr_buf);
+    ESPNOW_EXCEPTIONS(esp_now_del_peer(addr_buf));
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(espnow_del_peer_obj, espnow_del_peer);
+
+STATIC mp_obj_t espnow_send(mp_obj_t addr, mp_obj_t msg) {
+    mp_uint_t addr_len;
+    u8 *addr_buf;
+    mp_uint_t msg_len;
+    u8 *msg_buf = (u8 *)mp_obj_str_get_data(msg, &msg_len);
+    if (msg_len > ESP_NOW_MAX_DATA_LEN) mp_raise_ValueError("msg too long");
```

Comment:
how about "msg too long (%d bytes, but maximum is %d bytes)"?

Reply:
Looks fine too me.